### ISSUE TITLE
Profile page new mockup discussion

### ISSUE THREAD

**Issue by CleverFool77** [publiclab/plots2#6001]
#### Issues with the current design : 
- Overcrowds the page if a user has many subscriptions, similarly, the page looks blank for a user with no subscription. 
- How to display the user content with few research notes, wikis, questions but with no topics tagged to it or with no subscription. 
- A user might have interacted using the comments only, but comments have no feature of topics, therefore how to categorise them and display them?
- There are posts with more than one topics tagged to them, so if we categorise the posts according to topics, it will lead to repetition of content. Further increasing the complexity of the page.
-  Moreover, a user needs to search for his/her content in the tags card. A user always remembers the title of the wiki or the question instead of the tag under which it might lie. This creates a lot of confusion on the page.
- There is no feature for displaying the recent activity of a user or the most popular content of the user. 

@CleverFool77 and I have discussed a lot about this page and we feel that the current design where the content is neatly divided into Recent work, research notes, questions, comments etc is better for a good user experience. Therefore, we propose a mock up shown below : 

- Instead of classifying the content based on topics, it will be more feasible to classify them on the basis of research notes, questions, comments etc (like in the present design). Therefore, instead of tabs, we can have a dropdown of types. Similar to what we have done in the individual tags page. It makes the page more organised and user friendly. 

- The sidebar looks good, in addition to that, we can add the most active topics of the user on the sidebar. 

![Untitled (2)](https://user-images.githubusercontent.com/26685258/60942836-b47e9480-a301-11e9-8b29-36306695d7e7.png)

cc: @gautamig54 @jywarren

**Comment by CleverFool77** [comment]
Hi @gautamig54  I have added the mockup !! You can add the information and further details in this !!

**Comment by gautamig54** [comment]
Hi @CleverFool77!
If we are not categorising cards according to the most active topics, we can shift it to the sidebar, like we did for the related topics in the individual tags page. What say?

**Comment by CleverFool77** [comment]
I didn't do major change except for topic card. About shifting it to right and other stuff. I'm not sure. Maybe should we wait for Jeff's view? Then I'll make a new mockup. Till then, we can just list out the changes we want. And after all of them are listed. We'll just directly make a new mockup with all the listed required changes. What do you say?

**Comment by CleverFool77** [comment]
So the point you mentioned is one of the listed changes we might want.

**Comment by gautamig54** [comment]
Sure! I am writing down the points we discussed yesterday. Just few more minutes.

**Comment by CleverFool77** [comment]
Awesome !!! :tada:

**Comment by gautamig54** [comment]
@CleverFool77 I have edited the issue. Kindly have a look at it and add anything you feel is missing.

**Comment by CleverFool77** [comment]
You wrote almost everything !!! :tada: @gautamig54

**Comment by CleverFool77** [comment]
Hi @jywarren It would be great if you can look into this ..
Thanks !!!

**Comment by gautamig54** [comment]
Hi @CleverFool77. I has a discussion with liz about the design of profile page. The solution we came up with is : 
- We will display topic cards of 5-6 active topics of the user instead of showing all the topics a user is active or subscribed to. This will solve the issue of over crowding the page and also shows the most relevant activity of the user. Also, if we are showing just a few topic cards, then we can afford few repetitions.
-  Next, if a user has not subscribed to any topic or hasn't contributed to one, we will just display a message saying "you have not subscribed to any topic" and link the user to the topic or subscription page
- Now about comments or recent activity, we can have that option in the sidebar which will direct the user to a page consisting of the above content. 

cc : @jywarren

**Comment by CleverFool77** [comment]
Sure Gautami, I'll look into topic card portion then. And work on it. Right
now, I'm travelling a bit and have some interviews.
I'll be super active from monday once again.
Thanks !!

**Comment by jywarren** [comment]
Sounds awesome. Thanks!

**Comment by gautamig54** [comment]
Sure @CleverFool77! All the best for your interviews!

**Comment by CleverFool77** [comment]
Hi @gautmig54 @jywarren finished up with all stuff, now I'll be back to work.
Thanks!!!!!!

**Comment by jywarren** [comment]
Great! Welcome back! I'm publishing all the tag page work to live site now,
too!

**Comment by gautamig54** [comment]
The edit profile page is not yet fixed. I'll do it. 
@CleverFool77 there are a lot of cases where tags are concerned. Can we discuss it over call anytime you're free?

**Comment by ebarry** [comment]
Hi all!
Love the new profile page. I have two questions, 1) does anyone know why i (admin) would have experienced trouble adding the tag `new-york-city` to someone's profile page? , 2) why topics might be appearing multiple times on a single profile page? 

Here's a gif for #1) 
![adding-tags](https://user-images.githubusercontent.com/161439/63444135-04be3b80-c404-11e9-8265-2112484b651b.gif)

Here's a gif for #2) 
![extra-topics-shown](https://user-images.githubusercontent.com/161439/63444149-0d167680-c404-11e9-8b7b-5a7f02900226.gif)

**Comment by CleverFool77** [comment]
Hi @ebarry 
For First one, This is not the issue of Profile page. I guess there is some bug in tagging part !!!
For Second one, I'm not sure why there is repetition but I'll look into this issue.

**Comment by jywarren** [comment]
Also linking to ongoing work and discussion in: 

* https://github.com/publiclab/plots2/issues/6220
* https://github.com/publiclab/plots2/issues/6219

**Comment by jywarren** [comment]
Let's also:

* [ ] add a label to make it really clear (post #6219) what the collection of boxes on the profile page is -- right now it is `user.tags`, which is all tags on posts by the user. Should it instead be all tags the user is subscribed to? cc @steviepubliclab

**Comment by jywarren** [comment]
the last comment ☝️  really relates to some of the distinctions in this comment about the tag pages, too: https://github.com/publiclab/plots2/issues/6247#issuecomment-528578509

**Comment by steviepubliclab** [comment]
Hi all! Great work on the profile pages. I like the feed function, but one thing I have been struggling with is that the main feed often isn't featuring what I'm looking for-- for example, most of the time I'm looking for people's recent research notes, questions and comments, and not looking for things they've recently tagged. Wondering if we could alter the feed to feature those things and leave the tags used to that nice looking section on the top? 

For example, I got an email alert this morning that this person https://publiclab.org/profile/crispinpierce had commented on a question I had posted. So I visited their profile in hopes of finding other comments or posts they had recently made, but all I'm able to see from the front of their profile are tags which are already featured on the top of the page. 

Thanks all!!

**Comment by steviepubliclab** [comment]
on::
> Let's also:
> 
> * [ ]  add a label to make it really clear (post #6219) what the collection of boxes on the profile page is -- right now it is `user.tags`, which is all tags on posts by the user. Should it instead be all tags the user is subscribed to? cc @steviepubliclab

I think all the tags someone is subscribed to is nice. Might reduce the powertags there which are a bit confusing, see arrow below: 
<img width="967" alt="Screen Shot 2019-09-06 at 10 04 03 AM" src="https://user-images.githubusercontent.com/7490478/64434069-c2486000-d08d-11e9-976f-a8fb461230d8.png">

**Comment by jywarren** [comment]
OK cool, a couple options here:

1. we could screen out powertags in general
2. we could make the grey links in the right sidebar more obvious, or
convert them to a row across the top of the main column, that lead to that
author's comments, notes, and wiki edits
3. we could feature any note which the author has posted or commented on,
sorted by recency, within the topic cards in the main column, so you can
see recent posts, but in the context of the topic they belong to?

do any of these sound interesting and/or address some of the issues?
Thanks, Stevie!

**Comment by CleverFool77** [comment]
Hi !!! We get most used tags and notes according to that. But we can get the list of all notes through the links in sidebar. 
Hi @jywarren Regarding third point, it looks like good idea.
And regarding second point. Do you mean the convert them into row in like `23 ques 44 notes etc` above the topic cards ?

**Comment by CleverFool77** [comment]
Hi @jywarren Should these be converted to permanently blue with underline instead of turning blue on hover. 
So that we can know that its a link, and not just data which being shown.
As for a user who didn't about this, would think this as a text but not as a link. But by adding this color and underline. We would show it as a link.

![Screenshot from 2019-09-07 11-05-38](https://user-images.githubusercontent.com/26685258/64470355-9c8b8b80-d15f-11e9-817e-c9a6791af318.png)

Like this ?
![Screenshot from 2019-08-31 22-11-01](https://user-images.githubusercontent.com/26685258/64470360-aca36b00-d15f-11e9-8bb8-ac1bba121489.png)

**Comment by steviepubliclab** [comment]
Hi again! I'll reply to your comment @jywarren in line here:: 

1. we could screen out powertags in general
- Great! I think this would be a good move.

2. we could make the grey links in the right sidebar more obvious, or
convert them to a row across the top of the main column, that lead to that
author's comments, notes, and wiki edits
- nice. for some reason the right hand side of the page isn't where my eye goes to look for more material, but that's just me I think. Wondering what it would look like across the top? Either way making it more clear they are links would be helpful. 

3. we could feature any note which the author has posted or commented on,
sorted by recency, within the topic cards in the main column, so you can
see recent posts, but in the context of the topic they belong to?
- recent posts and comments sound good, but I'm not sure what you mean by "within the topic cards in the main column, so you can see recent posts, but in the context of the topic they belong to?" Can you clarify that part? 

Adding one just because I don't think it was captured in this set, -can we cut out the feed on tagging people do on profile pages? I think enough information about tags is captured in the top section and the tagging people do as a action isn't as important on the profile pages as content they create.

**Comment by jywarren** [comment]
I like the blue link text, thanks, @CleverFool77. Just to get to @steviepubliclab's request to see how it might look at the top of the main column, we could try something like this, of course substituting better-suited icons, and the icons could be in grey:

![image](https://user-images.githubusercontent.com/24359/64550388-d647c800-d300-11e9-8124-d2e76d94764a.png)

> "within the topic cards in the main column, so you can see recent posts, but in the context of the topic they belong to?"

Here I mean that the posts would be displayed within the topic cards, so they'd be grouped by topic. You can see an example of a topic card in the screenshot in this comment, too, and some example posts within that topic. For the profile, we'd list only posts to that topic by the profile author. 

Re: `the feed on tagging people do` - do you mean the area labelled "Most common tags"? Here, we have tried to make a space to describe (using topics) the content that people /post/ -- the tags weren't necessarily made by them but they are tags upon posts by the author. This gets at the distinction between "following" a tag and being "a contributor" to that tag. The listing here is a representation of their being a contributor to a range of topics. 

But as the discussion on https://github.com/publiclab/plots2/issues/6247 gets at, there is some work to disambiguate this. We can go in different directions here, but having this short list of tagnames is a standard and compact way to list some topics, so if there is a different short list of topics you'd like to feature here, it's available. But if we decide it's just too much information (especially if we move the sidebar listing of authored content into the main column and it gets crowded), we can just remove it. What do you think?

**Comment by steviepubliclab** [comment]
ok great. I like the links at the top with buttons to click on, but that's probably a personal preference. If it's busy with the "most common topics" section, maybe moving those to the side bar might make sense? just switching them I guess from what they currently are. Also, the phrasing there, it almost seems like "Most common topics" on PL rather than this person's profile. Wonder if we could clarify that this is "Most common topics _username_ shares about" or something like that? I'm not attached to that, just thinking out loud.   For naming and standardizing with the other thread, I'm good with whatever the group wants to go with. Here is sort of makes more sense to me to either do "Most common topics _username_ shares about" or "topics _username_ follows"

For the topic cards, would they bump up in order based on the most recent one someone had posted to? I think that would be good. Also, how do we choose which ones are featured?  For example, I could post the same RN with tags soil and agriculture, which would the profile page show -- hopefully not both? For example, this is what mine looks like right now, but the midwest stuff I post is also related to frac-sand. Not a great example, because one is a geography but you get the idea: 
<img width="506" alt="Screen Shot 2019-09-09 at 2 23 10 PM" src="https://user-images.githubusercontent.com/7490478/64556524-93d8b800-d30d-11e9-82a9-7be6d7b465bf.png">

For the feed on the tagging people do-- Maybe this isn't what happened, but it looked for a second  like whenever I went through and gardened adding tags to things, those things were showing up on my feed some thing like "stevie tagged _research note_ with _tag_."    Maybe not though?

**Comment by jywarren** [comment]
Just chiming in on "which is featured" -- we have a proposal (need to find
the issue) to have posts be shown only once (ah, here it is!
https://github.com/publiclab/plots2/issues/6199) and you're right that the
tag cards are ordered by recency. So if a post appears on a higher-up card,
it would be hidden in any lower cards even if it were cross-listed.

Like "by username" in Stevie's note, great!

Most common tags could swap positions with the grey list of content, that's
a good idea! We would want to keep the heading because that area is ALSO
used for profile tags, which is it's own can of worms. So this probably
needs a new mockup by now.

Did folks like the rough mockup of content-by-this-author I posted?

**Comment by ebarry** [comment]
I liked the rough mockup. 
i'm not sure if the icons are needed, but without them the horizontal layout might be hard to read, so i'm not sure on that. 

Would Activities refer to activities created as well as activities attempted?

**Comment by jywarren** [comment]
Not sure on that one. What do you think?

**Comment by ebarry** [comment]
what if we did not list Activities as a category? They would already be counted in notes.

**Comment by jywarren** [comment]
Sure and the notes display has a "type" filter too.

**Comment by steviepubliclab** [comment]
+1 on lumping activities in with notes. 

Just to clarify would the topic card only show the material the person who owns the profile posted or would it show posts from others who shared recently as well?

**Comment by jywarren** [comment]
Just the profile author's content, in this "version" of the topic card.
It's a variant of the typical card we show on other pages, customized just
for this purpose.

Thanks, we'll look at getting a main-column top bar with icons appropriate
for this. We'll also have to think about wrapping it to 2 lines on mobile
devices. That can be its own issue!
